### PR TITLE
Added elastic pooling to SQL Servers

### PR DESCRIPTION
**Pull request by Renetnielsen**
## Description

This introduces elastic pooling on SQL servers.
When this is introduced it will become a breaking change for SQL servers using this module.

### CODE REVIEW

File: azure/mssql-server/main.tf [hunk 1]
```diff
@@ -64,3 +64,24 @@ resource "azurerm_monitor_diagnostic_setting" "this" {
     }
   }
 }
+
+resource "azurerm_mssql_elasticpool" "this" {
+  name                = "mssqlpool-${lower(var.name)}-${lower(var.project_name)}-${lower(var.environment_short)}-${lower(var.environment_instance)}"
+  resource_group_name = var.resource_group_name
+  location            = var.location
+  server_name         = azurerm_mssql_server.this.name
+  license_type        = "LicenseIncluded"
```
Comment (djorgensendk) [inline azure/mssql-server/main.tf:73]: what is meant by license_type and "LicenseIncluded"?

File: azure/mssql-server/variables.tf [hunk 2]
```diff
@@ -77,6 +77,22 @@ variable log_retention_in_days {
   default     = 183
 }
 
+variable elastic_pool_sku {
+  type        = object({
+    name      = string
+    tier      = string
+    family    = string
+    capacity  = string
```
Comment (djorgensendk) [inline azure/mssql-server/variables.tf:85]: Based on how the module is used, capacity could be changed to a number

File: azure/mssql-server/main.tf (hunk 1 above)
Comment (Renetnielsen) [reply azure/mssql-server/main.tf:73]: `LicenseIncluded` means that Azure will add a License to the server.
`BasePrice` means that we have a license to use (So for instance if we bought it somewhere else).

**Review by djorgensendk — APPROVED** [review]

**Comment by Renetnielsen** [comment]
@cla-bot check

**Comment by kerbou** [comment]
Closing PR, based on latest comment. We can always re-open again

**Review by djorgensendk — APPROVED** [review]

**Pull request merged**